pkg/apis/v1alpha1/gcenodeclass.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2024 The CloudPilot AI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GCENodeClassSpec is the top level specification for the GCP Karpenter Provider.
+// This will contain the configuration necessary to launch instances in GCP.
+type GCENodeClassSpec struct {
```

Comment:
GCE or ce? 
https://cloud.google.com/products/compute?hl=en

What do you think?

Reply:
gce

Reply:
Google Compute Engine